### PR TITLE
netdev2: use params parameter for setup

### PR DESCRIPTION
**Pull request by OlegHahm**
For consistency and simplification reasons.

### CODE REVIEW

File: drivers/at86rf2xx/at86rf2xx.c [hunk 1]
```diff
@@ -35,23 +35,21 @@
...
     netdev2_t *netdev = (netdev2_t *)dev;
 
     netdev->driver = &at86rf2xx_driver;
     /* initialize device descriptor */
-    dev->spi = spi;
-    dev->cs_pin = cs_pin;
-    dev->int_pin = int_pin;
-    dev->sleep_pin = sleep_pin;
-    dev->reset_pin = reset_pin;
+    dev->spi = params->spi;
```
Comment (kaspar030) [inline drivers/at86rf2xx/at86rf2xx.c]: You could have a 'params' member of type at86rd2xx_params_t and just copy all at once.

**Comment by OlegHahm** [comment]
Addressed @kaspar030's comment.

File: drivers/at86rf2xx/at86rf2xx.c (hunk 1 above)
Comment (OlegHahm) [inline drivers/at86rf2xx/at86rf2xx.c]: Good idea.

**Comment by authmillenon** [comment]
netdev2_tap doesn't get this modification?

**Comment by OlegHahm** [comment]
`netdev2_tap_setup` has only one parameter.

**Comment by authmillenon** [comment]
How does it get the "tapx" string then?

**Comment by authmillenon** [comment]
Ah, that's the one parameter... got it.

**Comment by authmillenon** [comment]
But shouldn't it do this, too. Just for consistency?

File: drivers/at86rf2xx/at86rf2xx.c [hunk 2]
```diff
@@ -35,23 +35,17 @@
 #include "debug.h"
 
 
-void at86rf2xx_setup(at86rf2xx_t *dev, spi_t spi, spi_speed_t spi_speed,
-                     gpio_t cs_pin, gpio_t int_pin, gpio_t sleep_pin,
-                     gpio_t reset_pin)
+void at86rf2xx_setup(at86rf2xx_t *dev, at86rf2xx_params_t *params)
```
Comment (authmillenon) [inline drivers/at86rf2xx/at86rf2xx.c]: `const`?

File: drivers/encx24j600/encx24j600.c [hunk 3]
```diff
@@ -82,12 +82,12 @@ static inline void unlock(encx24j600_t *dev) {
     mutex_unlock(&dev->mutex);
 }
 
-void encx24j600_setup(encx24j600_t *dev, spi_t spi, gpio_t cs, gpio_t int_pin)
+void encx24j600_setup(encx24j600_t *dev, encx24j600_params_t *params)
```
Comment (authmillenon) [inline drivers/encx24j600/encx24j600.c]: `const`?

File: drivers/ethos/ethos.c [hunk 4]
```diff
@@ -47,16 +47,16 @@ static const uint8_t _esc_esc[] = {ETHOS_ESC_CHAR, (ETHOS_ESC_CHAR ^ 0x20)};
 static const uint8_t _esc_delim[] = {ETHOS_ESC_CHAR, (ETHOS_FRAME_DELIMITER ^ 0x20)};
 
 
-void ethos_setup(ethos_t *dev, uart_t uart, uint32_t baudrate, uint8_t *buf, size_t bufsize)
+void ethos_setup(ethos_t *dev, ethos_params_t *params)
```
Comment (authmillenon) [inline drivers/ethos/ethos.c]: `const`?

File: drivers/include/at86rf2xx.h [hunk 5]
```diff
@@ -143,11 +155,7 @@ typedef struct {
      * @brief   device specific fields
      * @{
      */
-    spi_t spi;                              /**< used SPI device */
-    gpio_t cs_pin;                          /**< chip select pin */
-    gpio_t sleep_pin;                       /**< sleep pin */
-    gpio_t reset_pin;                       /**< reset pin */
-    gpio_t int_pin;                         /**< external interrupt pin */
+    at86rf2xx_params_t params;              /**< parameters for initialization */
```
Comment (authmillenon) [inline drivers/include/at86rf2xx.h]: `const`?

File: drivers/xbee/xbee.c [hunk 6]
```diff
@@ -395,8 +395,7 @@ static int _set_proto(xbee_t *dev, uint8_t *val, size_t len)
 /*
  * Driver's "public" functions
  */
-int xbee_init(xbee_t *dev, uart_t uart, uint32_t baudrate,
-              gpio_t reset_pin, gpio_t sleep_pin)
+int xbee_init(xbee_t *dev, xbee_params_t *params)
```
Comment (authmillenon) [inline drivers/xbee/xbee.c]: `const`?

File: drivers/include/at86rf2xx.h [hunk 7]
```diff
@@ -181,9 +177,7 @@ typedef struct at86rf2xx_params {
  * @param[in] sleep_pin     GPIO pin connected to the sleep pin
  * @param[in] reset_pin     GPIO pin connected to the reset pin
```
Comment (authmillenon) [inline drivers/include/at86rf2xx.h]: Here and below: please adapt parameter doc

File: tests/driver_xbee/xbee_params.h [hunk 8]
```diff
@@ -27,7 +27,7 @@ static xbee_params_t xbee_params[] = {
     { .uart = XBEE_UART,
       .baudrate = 9600U,
       .sleep_pin = GPIO_UNDEF,
-      .status_pin = GPIO_UNDEF
+      .reset_pin = GPIO_UNDEF
```
Comment (authmillenon) [inline tests/driver_xbee/xbee_params.h]: Change unrelated?

File: drivers/at86rf2xx/at86rf2xx.c [hunk 9]
```diff
@@ -35,23 +35,17 @@
...
     netdev2_t *netdev = (netdev2_t *)dev;
 
     netdev->driver = &at86rf2xx_driver;
     /* initialize device descriptor */
-    dev->spi = spi;
-    dev->cs_pin = cs_pin;
-    dev->int_pin = int_pin;
-    dev->sleep_pin = sleep_pin;
-    dev->reset_pin = reset_pin;
+    dev->params = *params;
```
Comment (authmillenon) [inline drivers/at86rf2xx/at86rf2xx.c]: I think I would prefer a `memcpy()` here. And from some optimizations in the network stack I seem to remember that this is actually the better way to go ROM wise.

File: drivers/include/at86rf2xx.h (hunk 7 above)
Comment (OlegHahm) [inline drivers/include/at86rf2xx.h]: Wanted to wait if this approach gets approved first, will adapt.

File: tests/driver_xbee/xbee_params.h (hunk 8 above)
Comment (OlegHahm) [inline tests/driver_xbee/xbee_params.h]: Nope, there was a naming inconsistency.

File: drivers/at86rf2xx/at86rf2xx.c (hunk 9 above)
Comment (OlegHahm) [inline drivers/at86rf2xx/at86rf2xx.c]: Should be actually the same, but I will check.

Comment (OlegHahm) [inline drivers/at86rf2xx/at86rf2xx.c]: Interestingly the memcpy is -4 bytes on iotlab-m3 and +4 bytes on samr21-xpro.

Comment (authmillenon) [inline drivers/at86rf2xx/at86rf2xx.c]: Yeah, the M0 and M3 stuff have some weird differences... (maybe an optimization issue in GCC with M0?)

**Comment by OlegHahm** [comment]
addressed comments

File: drivers/at86rf2xx/at86rf2xx.c (hunk 9 above)
Comment (OlegHahm) [inline drivers/at86rf2xx/at86rf2xx.c]: If I compile with -O0 I get the same size for both version on samr21-xpro and still an improvement of 4 bytes with the memcpy version on iotlab.

File: drivers/include/ethos.h [hunk 10]
```diff
@@ -90,12 +100,9 @@ typedef struct {
...
  *
  * @param[out]  dev         handle of the device to initialize
- * @param[in]   uart        UART device to use
- * @param[in]   baudrate    baudrate for UART device
- * @param[in]   buf         buffer for incoming packets
- * @param[in]   bufsize     size of @p buf
+ * @param[in]   params      parameters for device initialization
  */
-void ethos_setup(ethos_t *dev, uart_t uart, uint32_t baudrate, uint8_t *buf, size_t bufsize);
+void ethos_setup(ethos_t *dev, ethos_params_t *params);
```
Comment (authmillenon) [inline drivers/include/ethos.h]: `const` ;-)

File: tests/driver_xbee/xbee_params.h (hunk 8 above)
Comment (authmillenon) [inline tests/driver_xbee/xbee_params.h]: Then please move into a seperate commit.

Comment (OlegHahm) [inline tests/driver_xbee/xbee_params.h]: like this?

Comment (authmillenon) [inline tests/driver_xbee/xbee_params.h]: Yapp :-)

**Comment by authmillenon** [comment]
ACK and go, when Murdock is happy (please squash).

**Comment by OlegHahm** [comment]
Why are the commits in the wrong order and why is the Murdock output empty?

**Comment by OlegHahm** [comment]
Hm, Travis agrees, Murdock shows green - even without output. Let's merge!

**Pull request merged**